core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -425,9 +422,17 @@ object H2OContext extends Logging {
     }
 
     override protected def getSparklingWaterHeartBeatEvent(): SparklingWaterHeartbeatEvent = {
-      val cloudV3 = getCloudInfo()
-      val memoryInfo = cloudV3.nodes.map(node => (node.ip_port, PrettyPrint.bytes(node.free_mem)))
-      SparklingWaterHeartbeatEvent(cloudV3.cloud_healthy, System.currentTimeMillis(), memoryInfo)
+      try {
+        val cloudV3 = getCloudInfo(conf)
+        val memoryInfo = cloudV3.nodes.map(node => (node.ip_port, PrettyPrint.bytes(node.free_mem)))
+        SparklingWaterHeartbeatEvent(cloudV3.cloud_healthy, System.currentTimeMillis(), memoryInfo)
+      } catch {
+        case e: H2OClusterNodeNotReachableException =>
+          H2OContext.get().head.stop()
+          throw new H2OClusterNodeNotReachableException(s"""
+External H2O cluster ${conf.h2oCluster.get} - ${conf.cloudName.get} is not reachable, H2OContext has been closed.
```

Comment:
nit: WDYT about reformatting the message?

Reply:
Yup, reformatted to be consistent